packages/reactivity/src/effect.ts

```diff
@@ -391,17 +391,15 @@ export function trigger(
 }
 
 export function triggerEffects(
-  dep: Dep | ReactiveEffect[],
+  dep: Dep,
   debuggerEventExtraInfo?: DebuggerEventExtraInfo
 ) {
-  // spread into array for stabilization
-  const effects = isArray(dep) ? dep : [...dep]
```

Comment:
I believe the reason for copying the `dep` into an array is to ensure that the subsequent iterations are stable. The calls to `triggerEffect()` inside those loops could modify `dep` while we're trying to iterate over it.

The PR description and linked issue don't seem to mention this, so I'm wondering whether there's been a misunderstanding about the purpose of the array?